analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -48,11 +48,17 @@ protected override void Initialize(SonarAnalysisContext context)
         private static void CheckTypesWithPrimaryConstructor(SonarSyntaxNodeReportingContext context)
         {
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
-            if (typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
+            if (!Excluded(typeDeclaration)
+                && typeDeclaration.ParameterList() is { Parameters.Count: 0 } parameterList
                 && !IsStructWithInitializedFieldOrProperty(typeDeclaration, context.SemanticModel))
             {
                 context.ReportIssue(Rule, parameterList, "primary constructor");
             }
+
+            static bool Excluded(TypeDeclarationSyntax node) =>
+                RecordDeclarationSyntaxWrapper.IsInstance(node)
```

Comment:
This is limiting the exception to records, but we have the same problem for classes.